presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,245 @@
+        }
+
+        for (Object value : exlucdeValues) {
+            builder.mustNot(new MatchQueryBuilder(columnName, getValue(type, value)));
+        }
+
+        if (includeValues.size() == 1) {
+            builder.must(new MatchQueryBuilder(columnName, getValue(type, getOnlyElement(includeValues))));
+        }
+        else {
+            for (Object value : includeValues) {
+                builder.should(new MatchQueryBuilder(columnName, getValue(type, value)));
+            }
+        }
+        return builder;
+    }
+
+    private QueryBuilder buildTermQuery(BoolQueryBuilder builder, String columnName, Domain domain, Type type)
+    {
+        List<Object> values = new ArrayList<>();
+        for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+            checkState(!range.isAll());
+            if (range.isSingleValue()) {
+                values.add(range.getLow().getValue());
+            }
+            else {
+                if (!range.getLow().isLowerUnbounded()) {
+                    switch (range.getLow().getBound()) {
+                        case ABOVE:
+                            builder.must(new RangeQueryBuilder(columnName).gt(getValue(type, range.getLow().getValue())));
+                            break;
+                        case EXACTLY:
+                            builder.must(new RangeQueryBuilder(columnName).gte(getValue(type, range.getLow().getValue())));
+                            break;
+                        case BELOW:
+                            throw new IllegalArgumentException("Low marker should never use BELOW bound");
+                        default:
+                            throw new AssertionError("Unhandled bound: " + range.getLow().getBound());
+                    }
+                }
+                if (!range.getHigh().isUpperUnbounded()) {
+                    switch (range.getHigh().getBound()) {
+                        case ABOVE:
+                            throw new IllegalArgumentException("High marker should never use ABOVE bound");
+                        case EXACTLY:
+                            builder.must(new RangeQueryBuilder(columnName).lte(getValue(type, range.getHigh().getValue())));
+                            break;
+                        case BELOW:
+                            builder.must(new RangeQueryBuilder(columnName).lt(getValue(type, range.getHigh().getValue())));
+                            break;
+                        default:
+                            throw new AssertionError("Unhandled bound: " + range.getHigh().getBound());
+                    }
+                }
+            }
+        }
+        if (values.size() == 1) {
+            builder.must(new TermQueryBuilder(columnName, getValue(type, getOnlyElement(values))));
+        }
+        return builder;
+    }
+
+    public static Object getValue(Type type, Object value)
+    {
+        if (type.equals(BIGINT)) {
+            return (long) value;
```

Comment:
We return `Object` from this method, so why are we casting `value` here? Casting seems redundant for `long`, `double`, and `boolean`. If it's for checking the type, then do `verify(x instanceOf ...)`.